### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by tahiry-dev**
Bubble sort is an algorithm that compares adjacent values and swaps them if necessary. 
Our project uses:  
- Variables
- Conditional Statements
- Functions
- Loops 
- Blocks
- Yield

### CODE REVIEW

File: bubble_main.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        swap = true
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+      end
+    end
+  end
+  new_arr = arr
+  puts "Sorted Array -> #{arr}"
+  print "\n"
+  new_arr
+end
```
Comment (bertil291utn) [inline bubble_main.rb]: ## Sort by

- Check this method

```ruby
unsorted=(1..10).to_a.reverse! 
%w[hi hey hello].bubble_sort_by(unsorted) { |left, right| left.length - right.length }
# => expected: ["hi", "hey", "hello"]
# => got: ["hi", "hello", "hey"]
     
```

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow you have a professional README file. 

However, in order to accomplish all the requirements , you required to do these changes, listed below:
- [x] Please fix linter issues
- [x] Elaborate README file, follow this [example file](https://github.com/TSHEPO-CLOUD/bubble_sort/tree/feature_branch#bubble_sort)

### Summary

-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

File: bubble_main.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      compare = yield(arr[i], arr[i + 1])
+      # puts compare
+      if compare == 1
+        swap = true
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+      end
+    end
+  end
+  arr
+end
```
Comment (peterrobert) [inline bubble_main.rb]: There is still a problem with this method kindly check below


`
       expected: ["hi", "hey", "hello"]
            got: ["hi", "hello", "hey"]
     `
### hint: kindly double check the implementation in your block

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 
### HELLO :wave: NICE WORK DOING THE CHANGES AS REQUESTED BY THE PREVIOUS TSE. HOWEVER, THERE IS STILL A PROBLEM IN ONE OF YOUR METHODS KINDLY CHECK BELOW :arrow_down: 

### IF YOU HAVE ANY QUESTION REACH ME ON SLACK @PETERROBERT

### HAPPY CODING :computer:

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Well done job :smile: 
You can merge it now :+1: 

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**